test/e2e/deploy_test.go

```diff
@@ -0,0 +1,126 @@
+package e2e_test
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pborman/uuid"
+
+	cli "github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/utils"
+	. "github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+)
+
+var _ = Describe("Deploy simple cluster", func() {
+
+	It("Release sample all-in-one.yaml should work", func() {
+		By("Prepare namespaces")
+		namespaceUserResources := "e2e-" + uuid.NewRandom().String() // TODO
+		namespaceOperator := "mongodb-atlas-kubernetes-system"
+		session := cli.Execute("kubectl", "create", "namespace", namespaceUserResources)
+		Expect(session).ShouldNot(Say("created"))
+		session = cli.Execute("kubectl", "create", "namespace", namespaceOperator)
+		Expect(session).ShouldNot(Say("created"))
+		userProjectConfig := cli.LoadUserProjectConfig("data/atlasproject.yaml")
+		userClusterConfig := cli.LoadUserClusterConfig("data/atlascluster_basic.yaml")
+
+		By("Check Kubernetes/MongoCLI version\n")
+		session = cli.Execute("kubectl", "version")
+		Eventually(session).Should(Say(K8sVersion))
+		session = cli.Execute("mongocli", "--version")
+		Eventually(session).Should(gexec.Exit(0))
+
+		By("Apply All-in-one configuration\n in ")
+		session = cli.Execute("kubectl", "apply", "-f", ConfigAll)
+		Eventually(session.Wait()).Should(Say("customresourcedefinition.apiextensions.k8s.io/atlasclusters.atlas.mongodb.com"))
+		Eventually(
+			cli.GetPodStatus(namespaceOperator),
+			"5m", "3s",
+		).Should(Equal("Running"))
+
+		By("Create secret")
+
+		session = cli.Execute("kubectl", "create", "secret", "generic", "my-atlas-key",
+			"--from-literal=orgId="+os.Getenv("MCLI_ORG_ID"),
+			"--from-literal=publicApiKey="+os.Getenv("MCLI_PUBLIC_API_KEY"),
+			"--from-literal=privateApiKey="+os.Getenv("MCLI_PRIVATE_API_KEY"),
+			"-n", namespaceUserResources,
+		)
+		Eventually(session.Wait()).Should(Say("my-atlas-key created"))
+
+		By("Create Sample Project\n")
+		session = cli.Execute("kubectl", "apply", "-f", ProjectSample, "-n", namespaceUserResources)
+		Eventually(session.Wait()).Should(Say("atlasproject.atlas.mongodb.com/my-project"))
+
+		By("Sample Cluster\n")
+		session = cli.Execute("kubectl", "apply", "-f", ClusterSample, "-n", namespaceUserResources)
+		Eventually(session.Wait()).Should(Say("atlascluster-sample created"))
+
+		By("Wait creating and check that it was created")
+		Eventually(cli.GetGeneration(namespaceUserResources)).Should(Equal("1"))
+		Eventually(
+			cli.IsProjectExist(userProjectConfig.Spec.Name),
+			"7m", "10s",
+		).Should(BeTrue())
+
+		projectID := cli.GetProjectID(userProjectConfig.Spec.Name)
```

Comment:
I'd recommend to get the project id from the `AtlasProject.status.id` instead of the mongocli. This will make sure that the Operator writes the correct id to the status

Reply:
hm... it probably could fix this:
a) project creating